engine/classes/Elgg/Router.php

```diff
@@ -95,17 +95,24 @@ public function route(\Elgg\Http\Request $request) {
 			$handled = call_user_func($function, $segments, $identifier);
 		}
 
-		$output = ob_get_clean();
+		if ($handled === true) {
+			$output = ob_get_clean();
+		} else if (is_string($handled)) {
+			$output = $handled;
+		}
```

Comment:
Sure. Should we maybe use a convenience function so we don't have to remember all the key names?